pkg/scheduler/plugins/elasticquota/plugin.go

```diff
@@ -0,0 +1,565 @@
+		klog.Warningf("update pod warning, update version for the same:%v", newPod.Name)
+		return
+	}
+
+	oldQuotaName := g.getPodAssociateQuotaName(oldPod)
+	newQuotaName := g.getPodAssociateQuotaName(newPod)
+	g.Lock()
+	defer g.Unlock()
+
+	// delete oldPod's Request
+	podReq, _ := resource.PodRequestsAndLimits(oldPod)
+	podReqTmp := quotav1.Subtract(corev1.ResourceList{}, podReq)
+	g.groupQuotaManager.UpdateGroupDeltaRequest(oldQuotaName, podReqTmp)
+
+	// add newPod's Request
+	podReq, _ = resource.PodRequestsAndLimits(newPod)
+	g.groupQuotaManager.UpdateGroupDeltaRequest(newQuotaName, podReq)
+
+	if oldQuotaName != newQuotaName {
+		if oldPod.Status.Phase == corev1.PodRunning {
+			if err := g.elasticQuotaInfos[oldQuotaName].deletePodIfPresent(oldPod); err != nil {
+				return
+			}
+			g.groupQuotaManager.UpdateGroupDeltaUsed(oldQuotaName, podReqTmp)
+			g.quotaOverUsedRevokeController.erasePodWithQuotaName(oldQuotaName, oldPod)
+		}
+		if newPod.Status.Phase == corev1.PodRunning {
+			if err := g.elasticQuotaInfos[newQuotaName].addPodIfNotPresent(newPod); err != nil {
+				return
+			}
+			g.groupQuotaManager.UpdateGroupDeltaUsed(newQuotaName, podReq)
+			g.quotaOverUsedRevokeController.addPodWithQuotaName(newQuotaName, newPod)
+		}
+	}
+
+	klog.V(3).Infof("OnPodUpdateFunc %v.%v add request success, quotaName:%v", newPod.Namespace, newPod.Name, newQuotaName)
+}
+
+func (g *Plugin) OnPodDelete(obj interface{}) {
+	<-g.failOver
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return
+	}
+
+	g.Lock()
+	defer g.Unlock()
+
+	quotaName := g.getPodAssociateQuotaName(pod)
+	podReq, _ := resource.PodRequestsAndLimits(pod)
+	podReqTmp := quotav1.Subtract(corev1.ResourceList{}, podReq)
+	g.groupQuotaManager.UpdateGroupDeltaRequest(quotaName, podReqTmp)
+
+	if err := g.elasticQuotaInfos[quotaName].deletePodIfPresent(pod); err != nil {
+		// the pod is not added to the g.elasticQuotaInfos[quotaName], so no need to call UpdateGroupDeltaUsed.
+		return
+	}
+	g.groupQuotaManager.UpdateGroupDeltaUsed(quotaName, podReqTmp)
+	g.quotaOverUsedRevokeController.erasePodWithQuotaName(quotaName, pod)
+	klog.V(3).Infof("OnPodDeleteFunc %v.%v delete request success", pod.Namespace, pod.Name)
+}
+
+func (g *Plugin) OnNodeAdd(obj interface{}) {
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		return
+	}
+
+	klog.V(4).Infof("OnNodeAddFunc add node[%v]", node.Name)
+	klog.V(6).Infof("OnNodeAddFunc add node: %+v", node)
+	if node.DeletionTimestamp != nil {
+		klog.Errorf("node is deleting:%v", node.Name)
+		return
+	}
+
+	allocatable := node.Status.Allocatable
+
+	g.nodeResourceMapLock.Lock()
+	defer g.nodeResourceMapLock.Unlock()
+	if _, ok := g.nodeResourceMap[node.Name]; ok {
+		klog.Infof("OnNodeAddFunc skip due OnAdd Twice:%v", node.Name)
+		return
+	} else {
+		g.nodeResourceMap[node.Name] = struct{}{}
+	}
+
+	g.groupQuotaManager.UpdateClusterTotalResource(allocatable)
+	klog.V(4).Infof("OnNodeAddFunc success:%v [%+v]", node.Name, allocatable)
```

Comment:
really need log the allocatable?

Reply:
deleted, the allocatable will log in the UpdateClusterTotalResource.